frameworks/helloworld/tests/test_sanity.py

```diff
@@ -284,6 +284,11 @@ def fn():
 
     shakedown.wait_for(lambda: fn())
 
+    # In order to prevent the second scheduler instance from obtaining a lock, we undo the "scale-up" operation
```

Comment:
I would place this after the asserts.

Additionally, you should add back the MARATHON_SINGLE_INSTANCE_APP label.